zhaquirks/sinope/switch.py

```diff
@@ -294,517 +372,460 @@ class SinopeTechnologiesMeteringCluster(CustomCluster, Metering):
+        cluster_id=SinopeManufacturerCluster.cluster_id,
+        enum_class=AlarmAction,
+        translation_key="alarm_options",
+        fallback_name="Alarm options",
+        entity_type=EntityType.CONFIG,
+    )
+    .enum(  # Flow alarm
+        attribute_name=SinopeManufacturerCluster.AttributeDefs.alarm_flow_threshold.name,
+        cluster_id=SinopeManufacturerCluster.cluster_id,
+        enum_class=FlowAlarm,
+        translation_key="alarm_flow",
+        fallback_name="Alarm flow",
+        entity_type=EntityType.CONFIG,
+    )
+    .enum(  # Abnormal Flow action
+        attribute_name=SinopeManufacturerCluster.AttributeDefs.abnormal_flow_action.name,
+        cluster_id=SinopeManufacturerCluster.cluster_id,
+        enum_class=AbnormalAction,
+        translation_key="abnormal_flow_action",
+        fallback_name="Abnormal flow action",
+        entity_type=EntityType.CONFIG,
+    )
+    .enum(  # Zone status
+        attribute_name=SinopeTechnologiesIasZoneCluster.AttributeDefs.zone_status.name,
+        cluster_id=SinopeTechnologiesIasZoneCluster.cluster_id,
+        enum_class=ZoneStatus,
+        translation_key="zone_status",
+        fallback_name="Zone status",
+        entity_type=EntityType.DIAGNOSTIC,
+    )
+    .enum(  # Emergency_power_source
+        attribute_name=SinopeManufacturerCluster.AttributeDefs.emergency_power_source.name,
+        cluster_id=SinopeManufacturerCluster.cluster_id,
+        enum_class=EmergencyPower,
+        translation_key="emergency_power_source",
+        fallback_name="Emergency power source",
+        entity_type=EntityType.CONFIG,
+    )
+    .enum(  # Valve status
+        attribute_name=SinopeTechnologiesMeteringCluster.AttributeDefs.status.name,
+        cluster_id=SinopeTechnologiesMeteringCluster.cluster_id,
+        enum_class=ValveStatus,
+        translation_key="valve_status",
+        fallback_name="Valve status",
+        entity_type=EntityType.DIAGNOSTIC,
+    )
+    .enum(  # Device status
+        attribute_name=SinopeManufacturerCluster.AttributeDefs.dev_status.name,
+        cluster_id=SinopeManufacturerCluster.cluster_id,
+        enum_class=DeviceStatus,
+        translation_key="dev_status",
+        fallback_name="Device status",
+        entity_type=EntityType.DIAGNOSTIC,
+    )
+    .sensor(  # battery percent
+        attribute_name=SinopeTechnologiesPowerConfigurationCluster.AttributeDefs.battery_percentage_remaining.name,
+        cluster_id=SinopeTechnologiesPowerConfigurationCluster.cluster_id,
+        state_class=SensorStateClass.MEASUREMENT,
+        unit=PERCENTAGE,
+        device_class=SensorDeviceClass.BATTERY,
+        reporting_config=ReportingConfig(
+            min_interval=30, max_interval=43200, reportable_change=1
+        ),
+        translation_key="battery_percentage_remaining",
+        fallback_name="Battery percentage remaining",
+        entity_type=EntityType.DIAGNOSTIC,
+    )
```

Comment:
Why is this needed? If you remove this, does the built-in ZHA battery percentage sensor not appear?